src/main/java/Task.java

```diff
@@ -0,0 +1,116 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
+    protected static int taskIndex = 0;
+    protected static final int MAX_ARRAY_SIZE = 100;
+    protected static Task[] taskArray = new Task[MAX_ARRAY_SIZE];
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public static void printList() {
+        int index = 0;
+        printTaskHeader();
+        for(Task task: taskArray) {
+            // Empty array
+            if(index >= taskIndex){
+                break;
+            }
+            task.printTaskIndex(index);
+            task.printTaskType();
+            task.printCompletionStatus();
+            task.printTaskItem();
+            System.out.println();
+            index++;
+        }
+        System.out.println("\n");
+    }
+
+    private static void printTaskHeader() {
+        String tasksTemplate
+                = "******************************************************\n"
+                + "*                                                    *\n"
+                + "*          [Objectives]-[Missions]-[Tasks]           *\n"
+                + "*                                                    *\n"
+                + "******************************************************\n";
+        System.out.println(tasksTemplate);
+    }
+
+    private void printTaskIndex(int taskIndex) {
+        System.out.print("[" + (taskIndex + 1) + "]");
+    }
+
+    private void printTaskType() {
+        System.out.print("[" + getTaskType() + "]");
+    }
+
+    protected String getTaskType() {
+        return null;
+    }
+
+    private void printCompletionStatus() {
+        System.out.print("[" + getStatusIcon() + "]");
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "\u2713" : "\u2718");
+    }
```

Comment:
I like how the used methods are subsequently ordered to increase the readability

Reply:
Thanks! I'll keep note on that!